lib/internal/Magento/Framework/App/Test/Unit/MaintenanceModeTest.php

```diff
@@ -92,10 +92,10 @@ public function testisOnWithIP()
             [MaintenanceMode::FLAG_FILENAME, true],
             [MaintenanceMode::IP_FILENAME, true],
         ];
-        $this->flagDir->expects($this->exactly(2))
+        $this->flagDir->expects($this->once())
             ->method('isExist')
             ->willReturnMap($mapisExist);
-        $this->assertFalse($this->model->isOn());
+        $this->assertTrue($this->model->isOn());
```

Comment:
I'm not sure if it's correct to change this, or if I should make the class behave like the old test expects.

The old test expects: when maintenance mode is enabled, and an allow-list file exists but is empty, then maintenance mode is off.

The new behaviour here feels like an improvement on this, but I might be missing something.

Reply:
Looks like now it's more correct. I would keep the new behavior